extensions/cdc-mysql/src/test/java/com/hazelcast/jet/cdc/mysql/MySqlCdcNetworkIntegrationTest.java

```diff
@@ -407,4 +404,27 @@ private static int findRandomOpenPortInRange(int fromInclusive, int toExclusive)
         throw new IOException("No free port in range [" + fromInclusive + ", " +  toExclusive + ")");
     }
 
+    @SuppressWarnings("StatementWithEmptyBody")
+    private static void assertJobFailsWithConnectException(Job job, boolean lenient) throws InterruptedException {
+        try {
+            //wait for job to finish w/ timeout
+            job.getFuture().get(5, SECONDS);
+        } catch (TimeoutException te) {
+            //explicitly cancelling the job because it has not completed so far
+            job.cancel();
+
+            if (lenient) {
+                //ignore the timeout; not all tests are deterministic, sometimes we don't end up in the state
+                //we actually want to test
+            } else {
+                fail("Connection failure not thrown");
+            }
+        } catch (ExecutionException ee) {
+            //job completed exceptionally, as expected, we check the details of it
+            Throwable rootCause = Throwables.getRootCause(ee);
+            assertEquals(JetException.class, rootCause.getClass());
```

Comment:
Better to use 

https://joel-costigliola.github.io/assertj/core/api/org/assertj/core/api/AbstractThrowableAssert.html#hasRootCauseInstanceOf(java.lang.Class)

than a shaded package `org.testcontainers.shaded.com.google.common.base.Throwables` from testcontainers